src/client/tile.cpp

```diff
@@ -1998,6 +2027,27 @@ static void apply_colorize(video::IImage *dst, v2u32 dst_pos, v2u32 size,
 }
 
 /*
+	Apply color to destination
+*/
+static void apply_paint(video::IImage *dst, v2u32 dst_pos, v2u32 size,
+		const video::SColor &color)
+{
+	video::SColor dst_c;
+
+	for (u32 y = dst_pos.Y; y < dst_pos.Y + size.Y; y++)
+	for (u32 x = dst_pos.X; x < dst_pos.X + size.X; x++) {
+		dst_c = dst->getPixel(x, y);
+		dst_c.set(
+				dst_c.getAlpha(),
+				MYMIN(255, dst_c.getRed() - (255 - color.getRed())),
```

Comment:
shouldn't this be clamped to the range `[0, 255]` instead of just enforcing a max of 255?